app/src/main/java/com/sopt/now/presentation/auth/login/LoginViewModel.kt

```diff
@@ -0,0 +1,107 @@
+package com.sopt.now.presentation.auth.login
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MutableLiveData
+import com.google.gson.Gson
+import com.sopt.now.data.auth.LoginData.LoginState
+import com.sopt.now.data.auth.LoginData.RequestLoginDto
+import com.sopt.now.data.auth.LoginData.ResponseLoginDto
+import com.sopt.now.data.auth.ServicePool
+import com.sopt.now.data.auth.User.ResponseUserDto
+import com.sopt.now.data.auth.User.UserState
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class LoginViewModel(application: Application) : AndroidViewModel(application) {
+    private val authService by lazy { ServicePool.authService }
+    val login_liveData = MutableLiveData<LoginState>()
+    val user_liveData = MutableLiveData<UserState>()
```

Comment:
```suggestion
    private var _loginLiveData = MutableLiveData<LoginState>()
    val loginLiveData : LiveData<LoginState> get() = _loginLiveData
    
    private var _userLiveData = MutableLiveData<UserState>()
    val userLiveData : LiveData<LoginState> get() = _userLiveData
```

`_가 붙은 것들`은 Mutable한 LiveData로, ViewModel 안에서만 수정 및 재할당이 가능하게 됩니다. 반면에 `_가 붙지 않은 것들`은 Immutable한 LiveData로, ViewModel을 바라보는 View 단에서 값을 읽기만 하게 합니다.(read-only)

이를 통해 ViewModel은 LiveData의 값을 수정 및 재할당하는 책임만을 갖게 되고, View는 LiveData의 값을 그저 읽기만 하면 되는 구조로 만들 수 있습니다.

기존의 코드라면, View의 코드에서도 충분히 LiveData의 값을 수정할 수 있습니다. 오로지 ViewModel만이 수정할 수 있게 한다면, 개발 과정 혹은 이후 이슈를 트래킹하는 과정에서 '어라 ... 어디서 이 LiveData의 값을 바꾸는거지 ...' 하면서 찾아 헤매는 수고를 덜 수 있습니다.